proto-public/pbmesh/v2beta1/api_gateway.proto

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+syntax = "proto3";
+
+package hashicorp.consul.mesh.v2beta1;
+
+import "pbresource/annotations.proto";
+import "pbresource/resource.proto";
+
+message APIGateway {
+  option (hashicorp.consul.resource.spec) = {scope: SCOPE_NAMESPACE};
+
+  // GatewayClassName is the name of the GatewayClass used by the MeshGateway
+  string gateway_class_name = 1;
+
+  // +kubebuilder:validation:MinItems=1
+  repeated APIGatewayListener listeners = 2;
+}
+
+message APIGatewayListener {
+  // Name is the name of the listener in a given gateway. This must be
+  // unique within a gateway.
+  string name = 1;
+
+  // +kubebuilder:validation:Maximum=65535
+  // +kubebuilder:validation:Minimum=0
+  uint32 port = 2;
+
+  // Protocol is the protocol that a listener should use, it must
+  // either be "http" or "tcp"
+  // +kubebuilder:validation≈ftg6:Enum=tcp,http
+  string protocol = 3;
+
+  // Hostname is the host name that a listener should be bound to, if
+  // unspecified, the listener accepts requests for all hostnames.
+  string hostname = 4;
+
+  // TLS is the TLS settings for the listener.
+  APIGatewayTLSConfiguration tls = 5;
+  //TODO NET-7311 @Gateway-Management
+}
+
+// APIGatewayTLSConfiguration specifies the configuration of a listener’s
+// TLS settings.
+message APIGatewayTLSConfiguration {
+  // Certificates is a set of references to certificates
+  // that a gateway listener uses for TLS termination.
+  repeated hashicorp.consul.resource.Reference certificates = 1;
+
+  // MaxVersion is the maximum TLS version that the listener
+  // should support.
+  optional string max_version = 2;
+
+  // MinVersion is the minimum TLS version that the listener
+  // should support.
+  optional string min_version = 3;
+
+  // Define a subset of cipher suites to restrict
+  // Only applicable to connections negotiated via TLS 1.2 or earlier
+  repeated string cipher_suites = 4;
```

Comment:
How well does [this pre-existing type](https://github.com/hashicorp/consul/blob/c2a0d4f9ca27aa69ec5555b2f92edff15e0d907c/proto-public/pbmesh/v2beta1/pbproxystate/transport_socket.proto#L110-L141) align with our needs for API gateway cipher suites and min/max TLS versions?

Reply:
Most likely we can just use them, I'll pull them in.

Reply:
The one thing they don't seem to have is certificates